### DISCUSSION TITLE
How can I say given packages A and B which share dependency C, which version of C is compatible with A and B?

### DISCUSSION THREAD

**Discussion by jrcii** [composer/composer#11374] [Q&A / Support] [answered]
I want to install `phppest/pest` and `laminas/laminas-test` which both depend on `phpunit/phpunit`. How can I automatically figure out which versions of pest and laminas-test I can install which will both be satisfied with the same version of phpunit?

**Comment by stof** [comment, marked as answer]
That's what composer does as part of its dependency resolution.

The easiest way is to require both packages together so that composer performs the constraint guessing for both packages as part of the same `require` command (if you do 2 successive require calls, the first one will choose a version without taking into account the second package, and the second require command might fail if it is incompatible)

**Reply by jrcii** [reply]
Thank you! This specifically was the answer:
```shell
composer require pestphp/pest:* laminas/laminas-test:* --dev -W
```
